modules/account-lib/src/coin/dot/transactionBuilder.ts

```diff
@@ -0,0 +1,384 @@
+    }
+    return this;
+  }
+  /**
+   *
+   * The nonce for this transaction.
+   *
+   * @param {number} nonce
+   * @returns {TransactionBuilder} This transaction builder.
+   *
+   * @see https://wiki.polkadot.network/docs/build-transaction-construction
+   */
+  sequenceId(nonce: sequenceId): this {
+    const value = new BigNumber(nonce.value);
+    this.validateValue(value);
+    this._nonce = value.toNumber();
+    return this;
+  }
+  /**
+   *
+   * The tip to increase transaction priority.
+   *
+   * @param {number | undefined} tip optional
+   * @returns {TransactionBuilder} This transaction builder.
+   *
+   * @see https://wiki.polkadot.network/docs/build-transaction-construction
+   */
+  tip(tip: number | undefined): this {
+    if (tip !== undefined) {
+      this.validateValue(new BigNumber(tip));
+    }
+    this._tip = tip;
+    return this;
+  }
+
+  /**
+   *
+   * The number of the checkpoint block after which the transaction is valid
+   *
+   * @param {number} blockNumber
+   * @returns {TransactionBuilder} This transaction builder.
+   *
+   * @see https://wiki.polkadot.network/docs/build-transaction-construction
+   */
+  validity({ firstValid }: Validity): this {
+    this.validateValue(new BigNumber(firstValid));
+    this._blockNumber = firstValid;
+    return this;
+  }
+  /**
+   *
+   * The hash of the checkpoint block.
+   *
+   * @param {number} blockHash
+   * @returns {TransactionBuilder} This transaction builder.
+   *
+   * @see https://wiki.polkadot.network/docs/build-transaction-construction
+   * @see https://wiki.polkadot.network/docs/build-protocol-info#transaction-mortality
+   */
+  blockHash(blockHash: string): this {
+    this._blockHash = blockHash;
+    return this;
+  }
+  /**
+   *
+   * The current version for transaction format.
+   *
+   * @param {number} transactionVersion
+   * @returns {TransactionBuilder} This transaction builder.
+   *
+   * @see https://wiki.polkadot.network/docs/build-transaction-construction
+   */
+  transactionVersion(transactionVersion: number): this {
+    this._transactionVersion = transactionVersion;
+    return this;
+  }
+
+  private method(method: TxMethod): this {
+    this._method = method;
+    return this;
+  }
+  /**
+   * Sets the
+   * specName,
+   * genesisHash - genesisHash of the chain,
+   * metadataRpc - The SCALE-encoded metadata for the runtime when submitted,
+   * specVersion - The current spec version for the runtime,
+   * chainName - chainName,
+   * registry,
+   *
+   * @returns {TransactionBuilder} This transaction builder.
+   *
+   * @see https://wiki.polkadot.network/docs/build-transaction-construction
+   */
+  testnet(): this {
+    this._specName = 'polkadot';
+    this._genesisHash = '0x2b8d4fdbb41f4bc15b8a7ec8ed0687f2a1ae11e0fc2dc6604fa962a9421ae349';
+    this._metadataRpc = metadataRpc;
+    this._specVersion = 9100;
+    this._chainName = 'Polkadot';
+    this.registry();
+    return this;
+  }
```

Comment:
I don't see this as hardcode data. 

- For static values:
Should  _coinConfig be the source of those values?
- For dynamica values:
Should it allow set those values from [/transaction/material endpoint data](https://paritytech.github.io/substrate-api-sidecar/dist/#operations-transaction-getTransactionMaterial)?